binding/HarfBuzzSharp.Shared/NativeObject.cs

```diff
@@ -1,90 +1,101 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Runtime.InteropServices;
 
 namespace HarfBuzzSharp
 {
 	public class NativeObject : IDisposable
 	{
-		private readonly bool zero;
+		private readonly bool _zero;
 
-		internal NativeObject(IntPtr handle)
+		private bool _isDisposed;
+
+		internal NativeObject (bool zero = true)
+			: this (IntPtr.Zero, zero)
 		{
-			Handle = handle;
-			zero = true;
 		}
 
-		internal NativeObject(IntPtr handle, bool zero)
+		internal NativeObject (IntPtr handle, bool zero = true)
 		{
 			Handle = handle;
-			this.zero = zero;
+			_zero = zero;
 		}
 
-		~NativeObject()
+		~NativeObject ()
 		{
-			Dispose(false);
+			Dispose (false);
 		}
 
 		public virtual IntPtr Handle { get; protected set; }
 
-		protected virtual void Dispose(bool disposing)
-		{
-			if (zero)
-				Handle = IntPtr.Zero;
-		}
-
-		public void Dispose()
+		public void Dispose ()
 		{
-			Dispose(true);
-			GC.SuppressFinalize(this);
+			Dispose (true);
+			GC.SuppressFinalize (this);
 		}
 
-		internal static int SizeOf<T>()
+		internal static int SizeOf<T> ()
 		{
 #if WINDOWS_UWP || NET_STANDARD
 			return Marshal.SizeOf<T>();
 #else
-			return Marshal.SizeOf(typeof(T));
+			return Marshal.SizeOf (typeof (T));
 #endif
 		}
 
-		internal static T PtrToStructure<T>(IntPtr intPtr)
+		internal static IntPtr StructureArrayToPtr<T> (IReadOnlyList<T> items)
 		{
-#if WINDOWS_UWP || NET_STANDARD
-			return Marshal.PtrToStructure<T>(intPtr);
-#else
-			return (T)Marshal.PtrToStructure(intPtr, typeof(T));
-#endif
+			var size = SizeOf<T> ();
+			var memory = Marshal.AllocCoTaskMem (size * items.Count);
+			for (var i = 0; i < items.Count; i++) {
+				var ptr = new IntPtr (memory.ToInt64 () + (i * size));
+				Marshal.StructureToPtr (items[i], ptr, true);
+			}
+			return memory;
 		}
 
-		internal static T[] PtrToStructureArray<T>(IntPtr intPtr, int count)
+		internal static IEnumerable<string> PtrToStringArray (IntPtr intPtr)
```

Comment:
This should return an array

Reply:
Probably. But I think this is fine since it is internal. The name is probably what is wrong here - we may use this to scan through a bigger lost and we might want to stop half way.